src/plugins/mf_priority.cpp

```diff
@@ -202,6 +202,30 @@ int process_nodelist (const std::string& nodelist_str) {
 }
 
 
+/*
+ * Helper function to extract the "type" and "count" key-value pairs
+ * from jobspec.
+ */
+int extract_resources (json_t *root) {
+    if (!root || !json_is_array (root)) return -1;
+
+    json_t *resources = json_array_get (root, 0);
+    if (!resources || !json_is_object (resources)) return -1;
+
+    json_t *type = json_object_get (resources, "type");
+    json_t *count = json_object_get (resources, "count");
+    if (!type || !json_is_string (type)) return -1;
+
+    // if the association did not explicitly request nodes for their job,
+    // just return 1
+    if (strcmp (json_string_value (type), "node") != 0) return 1;
+
+    if (!count || !json_is_integer (count)) return -1;
+    int nnodes = json_integer_value (count);
+    return nnodes;
+}
```

Comment:
Let's consider how to do this more reliably. Probably returning 1 node if a user doesn't explicitly request nodes is not going to work in production because users could bypass all node limits by always requesting only cores:  e.g. `flux alloc -n64000`

Instead we may have to do something more like the flux-core [limit-job-size](https://github.com/flux-framework/flux-core/blob/master/src/modules/job-manager/plugins/limit-job-size.c) plugin, where there is a limit on the total number of cores+nodes+gpus and if a job exceeds any one of those it is deferred.